crates/config/src/config.rs

```diff
@@ -109,26 +109,81 @@ pub struct ChallengerConfig {
     pub burn_lock: Script,
 }
 
+#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
+pub struct ContractTypeScriptConfig {
+    pub state_validator: Script,
+    pub deposit_lock: Script,
+    pub stake_lock: Script,
+    pub custodian_lock: Script,
+    pub withdrawal_lock: Script,
+    pub challenge_lock: Script,
+    pub l1_sudt: Script,
+    pub allowed_eoa_scripts: HashMap<H256, Script>,
+    pub allowed_contract_scripts: HashMap<H256, Script>,
+}
+
+#[derive(Clone, Debug)]
+pub struct ContractsCellDep {
+    pub rollup_cell_type: CellDep,
+    pub deposit_cell_lock: CellDep,
+    pub stake_cell_lock: CellDep,
+    pub custodian_cell_lock: CellDep,
+    pub withdrawal_cell_lock: CellDep,
+    pub challenge_cell_lock: CellDep,
+    pub l1_sudt_type: CellDep,
+    pub allowed_eoa_locks: HashMap<H256, CellDep>,
+    pub allowed_contract_types: HashMap<H256, CellDep>,
+}
+
 #[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
 pub struct BlockProducerConfig {
     pub account_id: u32,
     #[serde(default = "default_check_mem_block_before_submit")]
     pub check_mem_block_before_submit: bool,
     // cell deps
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub rollup_cell_type_dep: CellDep,
     pub rollup_config_cell_dep: CellDep,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub deposit_cell_lock_dep: CellDep,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub stake_cell_lock_dep: CellDep,
     pub poa_lock_dep: CellDep,
     pub poa_state_dep: CellDep,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub custodian_cell_lock_dep: CellDep,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub withdrawal_cell_lock_dep: CellDep,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub challenge_cell_lock_dep: CellDep,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub l1_sudt_type_dep: CellDep,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub allowed_eoa_deps: HashMap<H256, CellDep>,
+    #[deprecated]
+    #[serde(skip_serializing)]
+    #[serde(default)]
     pub allowed_contract_deps: HashMap<H256, CellDep>,
     pub challenger_config: ChallengerConfig,
     pub wallet_config: WalletConfig,
+    #[serde(default)]
+    pub contract_type_scripts: ContractTypeScriptConfig,
```

Comment:
This part is belongs to consensus, we I think maybe we can move it out from BlockProducer, and add a new consensus config?

``` rust
pub struct ConsensusConfig {
  pub contract_type_scripts: ContractTypeScriptConfig,
}
```